test/perf/regression/target.go

```diff
@@ -0,0 +1,265 @@
+package regression
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+	"strings"
+
+	"k8s.io/apimachinery/pkg/runtime/schema"
+
+	"text/template"
+
+	"github.com/ghodss/yaml"
+	"github.com/sirupsen/logrus"
+	batchv1 "k8s.io/api/batch/v1"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/watch"
+	"k8s.io/client-go/discovery"
+	"k8s.io/client-go/dynamic"
+	"k8s.io/client-go/kubernetes"
+)
+
+type target struct {
+	typedClient     kubernetes.Interface
+	discoveryClient discovery.DiscoveryInterface
+	untypedClient   dynamic.Interface
+	namespacePrefix string
+	logger          *logrus.Logger
+	resourcePaths   []string
+	etcdImage       string
+	etcdbrImage     string
+	namespace       string
+}
+
+func (t *target) setup() error {
+	t.logger.Info("Setting up target...")
+
+	if strings.TrimSpace(t.etcdImage) == "" || strings.TrimSpace(t.etcdbrImage) == "" {
+		return fmt.Errorf("Cannot test with invalid images. etcdImage: %s, etcdbrImage: %s", t.etcdImage, t.etcdbrImage)
+	}
+
+	err := t.createNamespace()
+	if err != nil {
+		return err
+	}
+
+	err = t.createResources()
+	if err != nil {
+		return err
+	}
+
+	return err
+}
+
+func (t *target) teardown() {
+	t.logger.Info("Tearing down target...")
+
+	t.logger.Infof("Deleting the namespace %s", t.namespace)
+	err := t.typedClient.CoreV1().Namespaces().Delete(t.namespace, &metav1.DeleteOptions{})
+	if err != nil {
+		t.logger.Errorf("Error deleting namespace %s: %s", t.namespace, err)
+		return
+	}
+
+	t.logger.Infof("Deleted the namespace %s", t.namespace)
+}
+
+func (t *target) createNamespace() error {
+	t.logger.Infof("Creating namespace with prefix %s", t.namespacePrefix)
+	ns, err := t.typedClient.CoreV1().Namespaces().Create(&corev1.Namespace{
+		ObjectMeta: metav1.ObjectMeta{
+			GenerateName: t.namespacePrefix,
+		},
+	})
+
+	if err != nil {
+		return err
+	}
+
+	t.namespace = ns.Name
+	t.logger.Infof("Created namespace %s", t.namespace)
+	return nil
+}
+
+func (t *target) createResources() error {
+	t.logger.Infof("Creating resources in the namespace %s", t.namespace)
+	params := map[string]string{
+		"namespace":   t.namespace,
+		"etcdImage":   t.etcdImage,
+		"etcdbrImage": t.etcdbrImage,
+	}
+	for _, resourcePath := range t.resourcePaths {
+		t.logger.Infof("Creating resource from %s", resourcePath)
+
+		var (
+			u   *unstructured.Unstructured
+			gvr *schema.GroupVersionResource
+			ri  dynamic.ResourceInterface
+			err error
+		)
+		{
```

Comment:
Why this scope?

Reply:
To release `renderedJSON` for GC as soon as possibly.